framework/web/Controller.php

```diff
@@ -261,4 +311,27 @@ public function refresh($anchor = '')
     {
         return Yii::$app->getResponse()->redirect(Yii::$app->getRequest()->getUrl() . $anchor);
     }
+
+    /**
+     * Send data formatted as specified.
+     *
+     * This method is a shortcut for sending data formatted as specified by the `$format` parameter.
+     * It will return the [[Application::getResponse()|response]] application component after configuring
+     * the [[Response::$format|format]] and setting the [[Response::$data|data]] that should be formatted.
+     * Used internally by the [[Controller::asJson()]], [[Controller::asXml()]] and similar methods.
+     *
+     * @param string $format the format `$data` should be formatted as.
+     * @param mixed $data the data that should be formatted.
+     * @return Response a response that is configured to send `$data` formatted as specified.
+     * @since 2.0.12
+     * @see Response::$format
+     * @see ResponseFormatterInterface
+     */
+    private function formatData($format, $data)
```

Comment:
I suggest make this method protected - it could be used to handle custom response formats, so developer could create own `as*()` methods easily.
It could be also named more descriptive, for example `setResponseData($data, $format = Response::FORMAT_RAW)`